### PR TITLE
Fixed issue #6458

### PR DESCRIPTION
**Pull request by akathuria289**
## Issue Reference
#6458

### CODE REVIEW

**Comment by Sanjaybhattwebkul** [comment]
@akathuria289 
The product's **price** can be a **decimal** value. so it is still a bug.
Please find the screenshot [Here](https://prnt.sc/whH2OJLO2RYK)

**Comment by akathuria289** [comment]
@Sanjaybhattwebkul Fixed

File: packages/Webkul/Admin/src/Resources/views/marketing/promotions/catalog-rules/create.blade.php [hunk 1]
```diff
@@ -218,8 +218,9 @@
                     </div>
 
                     <div v-else>
-                        <div class="control-group" v-if="matchedAttribute.type == 'text' || matchedAttribute.type == 'price' || matchedAttribute.type == 'decimal' || matchedAttribute.type == 'integer'">
-                            <input class="control" :name="['conditions[' + index + '][value]']" v-model="condition.value"/>
+                        <div class="control-group" :class="[errors.has('value') ? 'has-error' : '']" v-if="matchedAttribute.type == 'text' || matchedAttribute.type == 'price' || matchedAttribute.type == 'decimal' || matchedAttribute.type == 'integer'">
+                            <input v-validate="matchedAttribute.type == 'price'?'decimal:2':'' || matchedAttribute.type == 'decimal'?'decimal:2':'' || matchedAttribute.type == 'integer' ? 'decimal:2':'' || matchedAttribute.type == 'text'?'alpha_num':''" class="control" :name="['conditions[' + index + '][value]']" v-model="condition.value" name="value"/>
+                            <span class="control-error" v-if="errors.has('value')">@{{ errors.first('value') }}</span>
```
Comment (devansh-webkul) [inline packages/Webkul/Admin/src/Resources/views/marketing/promotions/catalog-rules/create.blade.php:223]: Use `v-text` here.

File: packages/Webkul/Admin/src/Resources/views/marketing/promotions/catalog-rules/create.blade.php [hunk 2]
```diff
@@ -218,8 +218,9 @@
                     </div>
 
                     <div v-else>
-                        <div class="control-group" v-if="matchedAttribute.type == 'text' || matchedAttribute.type == 'price' || matchedAttribute.type == 'decimal' || matchedAttribute.type == 'integer'">
-                            <input class="control" :name="['conditions[' + index + '][value]']" v-model="condition.value"/>
+                        <div class="control-group" :class="[errors.has('value') ? 'has-error' : '']" v-if="matchedAttribute.type == 'text' || matchedAttribute.type == 'price' || matchedAttribute.type == 'decimal' || matchedAttribute.type == 'integer'">
+                            <input v-validate="matchedAttribute.type == 'price'?'decimal:2':'' || matchedAttribute.type == 'decimal'?'decimal:2':'' || matchedAttribute.type == 'integer' ? 'decimal:2':'' || matchedAttribute.type == 'text'?'alpha_num':''" class="control" :name="['conditions[' + index + '][value]']" v-model="condition.value" name="value"/>
```
Comment (devansh-webkul) [inline packages/Webkul/Admin/src/Resources/views/marketing/promotions/catalog-rules/create.blade.php:222]: Proper spacing.

**Review by devansh-webkul — CHANGES_REQUESTED** [review]

**Review by devansh-webkul — APPROVED** [review]

File: packages/Webkul/Admin/src/Resources/views/marketing/promotions/catalog-rules/create.blade.php [hunk 3]
```diff
@@ -218,8 +218,9 @@
                     </div>
 
                     <div v-else>
-                        <div class="control-group" v-if="matchedAttribute.type == 'text' || matchedAttribute.type == 'price' || matchedAttribute.type == 'decimal' || matchedAttribute.type == 'integer'">
-                            <input class="control" :name="['conditions[' + index + '][value]']" v-model="condition.value"/>
+                        <div class="control-group" :class="[errors.has('value') ? 'has-error' : '']" v-if="matchedAttribute.type == 'text' || matchedAttribute.type == 'price' || matchedAttribute.type == 'decimal' || matchedAttribute.type == 'integer'">
+                            <input v-validate="matchedAttribute.type == 'price' ? 'decimal:2' : '' || matchedAttribute.type == 'decimal' ? 'decimal:2' : '' || matchedAttribute.type == 'integer' ? 'decimal:2' : '' || matchedAttribute.type == 'text' ? 'alpha_num' : ''" class="control" :name="['conditions[' + index + '][value]']" v-model="condition.value" name="value"/>
```
Comment (jitendra-webkul) [inline packages/Webkul/Admin/src/Resources/views/marketing/promotions/catalog-rules/create.blade.php:222]: add space after decimal:

**Pull request merged**